src/controllers/movie.js

```diff
@@ -0,0 +1,167 @@
+const { Prisma } = require('@prisma/client');
+const prisma = require('../utils/prisma');
+const getErorCode = (e, res) => {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+            return res.status(409).json({ error: "A movie with the provided field already exists" })
+        } else if (e.code === "P2025") {
+            return res.status(404).json({ error: "A movie with the provided id does not exists!" })
+        }
+    }
+    res.status(500).json({ error: e.code + " " + e.message })
+}
```

Comment:
I like this. I've noticed this function is repeated in the various different controller files though, I think you could figure out a way to make this a single function that all of your controllers can use.

You might have some kind of error map object that looks something like:

```js
const ERROR_MESSAGES = {
    movies: {
        P2002: "A movie with the provided field already exists"
    }
}

getErrorCode(e, res, 'movies')
```